enarx-keep/tests/bin/read.c

```diff
@@ -0,0 +1,22 @@
+#include "libc.h"
+
+int main(void) {
+    int min(int a, int b) { return a < b ? a : b; }
+
+    char buf[12] = {'\0'};
+    int len = 1;
+    int bytes = 0;
+
+    while (1) {
+        bytes = read(STDIN_FILENO, buf, len);
+
+        if (bytes > 0) {
+           write(STDOUT_FILENO, buf, bytes);
+        } else
+            break;
+
+        len = min(len * 2, 12);
+    }
+
+    return bytes;
+}
```

Comment:
```suggestion
int min(int a, int b) {
    return a < b ? a : b;
}

int main() {
    char buf[16] = {};

    for (size_t in = 1;; in = min(in * 2, sizeof(buf)))  {
        ssize_t out = read(STDIN_FILENO, buf, in);
        if (out <= 0)
            break;

        write(STDOUT_FILENO, buf, out);
    }
}
```

Reply:
Wow - I didn't realize you could leave the middle test expression out.